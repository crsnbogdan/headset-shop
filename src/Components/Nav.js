import AddIcon from '@mui/icons-material/Add';
import CloseIcon from '@mui/icons-material/Close';
import DeleteIcon from '@mui/icons-material/Delete';
import MenuIcon from '@mui/icons-material/Menu';
import RemoveIcon from '@mui/icons-material/Remove';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import { AnimatePresence, motion } from 'framer-motion';
import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import uniqid from 'uniqid';
import '../Styles/Nav.css';

const Nav = (props) => {
  const [mobileNavVisible, setMobileNavVisible] = useState(false);

  let cartProducts = props.cart.map((cartItem) => {
    return (
      <motion.div
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        transition={{ duration: 0.25 }}
        exit={{ opacity: 0, display: 'none' }}
        className="nav__cartitem grid p-2 bg-white"
        key={uniqid()}
      >
        <img className="cartitem__img" src={cartItem.product.image} alt="" />
        <div className="cartitem__info flex flex-col justify-between text-sm text-black">
          <p className="text-black font-semibold cartitem_itemtotal">
            <span className="font-normal">Total: </span>
            {(
              cartItem.qty *
              (cartItem.product.salePrice || cartItem.product.regularPrice)
            ).toFixed(2)}
            $
          </p>
          <p className="text-sm">
            {cartItem.product.name.substring(0, 40).trim() + '(...)'}
          </p>
          <p className="cartitem__qtyval">Quantity: {cartItem.qty}</p>
        </div>
        <div className="cartitem__actions justify-between items-center flex flex-col">
          <DeleteIcon
            className="cartitem__delete"
            onClick={() => {
              props.setCartQuantity(props.cartQuantity - cartItem.qty);

              let productInCartIndex = props.cart.findIndex(
                (item) => cartItem.sku === item.sku
              );
              let cachedCart = props.cart;
              cachedCart.splice(productInCartIndex, 1);
              props.setCart([...cachedCart]);
            }}
          />
          <div className="cartitem__qtymodifiers flex items-center ">
            <AddIcon
              className="cartitem__qtymodifier add"
              onClick={() => {
                props.setCartQuantity(props.cartQuantity + 1);
                let productInCartIndex = props.cart.findIndex(
                  (item) => cartItem.sku === item.sku
                );
                let cachedCart = props.cart;
                cachedCart[productInCartIndex].qty += 1;
                props.setCart([...cachedCart]);
              }}
            />
            <p className="text-lg text-black font-semibold py-1 px-3 bg-slate-100">
              {cartItem.qty}
            </p>
            <RemoveIcon
              className="cartitem__qtymodifier subtract"
              onClick={() => {
                props.setCartQuantity(props.cartQuantity - 1);
                let productInCartIndex = props.cart.findIndex(
                  (item) => cartItem.sku === item.sku
                );
                let cachedCart = props.cart;
                cachedCart[productInCartIndex].qty -= 1;
                if (cachedCart[productInCartIndex].qty === 0) {
                  cachedCart.splice(productInCartIndex, 1);
                  if (cachedCart.length === 0) {
                    props.setCartVisible(false);
                  }
                  return props.setCart([...cachedCart]);
                }
                props.setCart([...cachedCart]);
              }}
            />
          </div>
        </div>
      </motion.div>
    );
  });

  return (
    <div className="nav__container">
      <nav className="nav items-center relative text-white flex px-28 py-6 justify-between">
        <h3 className="text-2xl text-yellow-300 font-semibold">
          KB<span className="text-white font-light text-lg">.shop</span>
        </h3>
        <MenuIcon
          className="nav__sidebartoggle"
          fontSize="large"
          onClick={() => {
            setMobileNavVisible(true);
            document
              .querySelector('.page__navoverlay')
              .classList.toggle('visible');
          }}
        />
        <motion.div
          initial={{ right: -300 }}
          animate={{
            right: mobileNavVisible ? 0 : -300,
          }}
          transition={{ duration: 0.5 }}
          className="nav__links w-3/12 text-lg flex items-center justify-between"
        >
          <CloseIcon
            className="nav__close"
            fontSize="large"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          />
          <NavLink
            to="/"
            end
            className="nav__link cart"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          >
            home
          </NavLink>
          <NavLink
            to="shop"
            end
            className="nav__link"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          >
            shop
          </NavLink>
          <NavLink
            to="cart"
            className="nav__link relative"
            onClick={() => {
              setMobileNavVisible(false);
              document
                .querySelector('.page__navoverlay')
                .classList.toggle('visible');
            }}
          >
            {props.cartQuantity > 0 ? (
              <motion.div
                initial={{ scale: 0.5 }}
                animate={{ scale: 1 }}
                transition={{ duration: 0.25 }}
                className="nav__qty flex justify-center items-center color-white absolute text-xs font-medium rounded-full"
              >
                <p className="text-white">{props.cartQuantity}</p>
              </motion.div>
            ) : null}
            <ShoppingCartIcon
              fontSize="medium"
              sx={{ padding: '10px', boxSizing: 'content-box' }}
              onMouseEnter={() => props.setCartVisible(true)}
              onMouseLeave={() => props.setCartVisible(false)}
            />
          </NavLink>
        </motion.div>
        <AnimatePresence>
          {props.cartVisible && (
            <motion.div
              initial={{ minHeight: '0px' }}
              animate={{ minHeight: '100px' }}
              exit={{ minHeight: '0px' }}
              transition={{ duration: 0.5, delayChildren: 1 }}
              onMouseEnter={() => props.setCartVisible(true)}
              onMouseLeave={() => props.setCartVisible(false)}
              className="nav__cartcontainer p-2 bg-slate-100 absolute"
            >
              {props.cart.length === 0 && props.cartVisible && (
                <motion.p
                  initial={{ display: 'none' }}
                  animate={{ display: 'block' }}
                  exit={{ display: 'none' }}
                  transition={{ delay: 0.2 }}
                  className="nav__cartemptytext text-slate-400"
                >
                  The cart is empty
                </motion.p>
              )}
              {props.cart && props.cartVisible && cartProducts}
              {props.cart.length > 0 && props.cartVisible && (
                <motion.p
                  exit={{ display: 'none' }}
                  className="text-black nav__carttotal"
                >
                  Total:{' '}
                  <span className="font-semibold">
                    {props.cart
                      .reduce(
                        (total, product) =>
                          total +
                          product.qty *
                            (product.product.salePrice ||
                              product.product.salePrice),
                        0
                      )
                      .toFixed(2)}
                    $
                  </span>
                </motion.p>
              )}
            </motion.div>
          )}
        </AnimatePresence>
      </nav>
    </div>
  );
};

export default Nav;
