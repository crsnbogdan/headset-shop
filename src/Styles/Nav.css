nav {
    background: linear-gradient(180deg, rgb(13, 23, 45) 15%, rgb(16, 9, 61) 100%);
    height: 10vh;
    border-bottom: 2px solid rgb(253,224, 71);
}

.nav__link {
    color: white;
}
.nav__link span {
    color: #fde047;
}
.nav__link.active {
    color: #fde047;
}

#cart__underline {
    width: 0%;
    height: 2px;
    background-color: red;
}

.nav__link {
    border-bottom: 0px solid #fde047;
    transition: 0.5s ease-in;
}
.nav__link.active {
    border-bottom: 2px solid #fde047;
}

.nav__qty {
 pointer-events: none;
 background-color: red;
 width: 20px;
 height: 20px;
 top: -6px;
 right: -4px;
}

@media screen and (max-width: 1024px) {
    nav .nav__links  {
        width: 50%!important;
    }
}
@media screen and (max-width: 550px) {
    .nav {
        padding: 20px 40px!important;
    }
    .nav .nav__links  {
        width: 50%!important;
    }
}