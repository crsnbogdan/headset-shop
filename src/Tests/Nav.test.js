import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ReactDOM from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../Components/Nav';
let cart = [
  {
    product: {
      name: 'Product 1',
      customerReviewAverage: 4.7,
      customerReviewCount: 2267,
      regularPrice: 99.99,
      salePrice: 89.99,
      longDescription: 'This is a test',
      image:
        'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg',
      sku: 1,
    },
    sku: 1,
    qty: 1,
  },
  {
    product: {
      name: 'Product 2',
      customerReviewAverage: 4.7,
      customerReviewCount: 2267,
      regularPrice: 99.99,
      salePrice: 89.99,
      longDescription: 'This is a test',
      image:
        'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg',
      sku: 2,
    },
    sku: 2,
    qty: 4,
  },
  {
    product: {
      name: 'Product 3',
      customerReviewAverage: 3.7,
      customerReviewCount: 100,
      regularPrice: 95.99,
      salePrice: 79.99,
      longDescription: 'This is a test',
      image:
        'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/6298/6298657_sd.jpg',
      sku: 3,
    },
    sku: 3,
    qty: 2,
  },
];

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
  cart = [
    {
      product: {
        name: 'Product 1',
        customerReviewAverage: 4.7,
        customerReviewCount: 2267,
        regularPrice: 99.99,
        salePrice: 89.99,
        longDescription: 'This is a test',
        image:
          'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg',
        sku: 1,
      },
      sku: 1,
      qty: 1,
    },
    {
      product: {
        name: 'Product 2',
        customerReviewAverage: 4.7,
        customerReviewCount: 2267,
        regularPrice: 99.99,
        salePrice: 89.99,
        longDescription: 'This is a test',
        image:
          'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg',
        sku: 2,
      },
      sku: 2,
      qty: 4,
    },
    {
      product: {
        name: 'Product 3',
        customerReviewAverage: 3.7,
        customerReviewCount: 100,
        regularPrice: 95.99,
        salePrice: 79.99,
        longDescription: 'This is a test',
        image:
          'https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/6298/6298657_sd.jpg',
        sku: 3,
      },
      sku: 3,
      qty: 2,
    },
  ];
});

it('renders correctly and matches snapshot', () => {
  const setCartQuantity = jest.fn();
  const setCart = jest.fn();
  const setCartVisible = jest.fn();

  const navTree = renderer
    .create(
      <BrowserRouter>
        <Nav
          cart={cart}
          cartQuantity={0}
          cartVisible={true}
          setCart={setCart}
          setCartQuantity={setCartQuantity}
          setCartVisible={setCartVisible}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(navTree).toMatchSnapshot();
});

it('shows products present in props.cart in the cart', () => {
  act(() => {
    ReactDOM.createRoot(container).render(
      <BrowserRouter>
        <Nav cart={cart} cartVisible={true} />
      </BrowserRouter>
    );
  });
  expect(container.querySelector('.nav__cartcontainer')).toBeInTheDocument();
  let cartItems = [...container.querySelectorAll('.nav__cartitem')];
  expect(cartItems.length).toEqual(cart.length);
});

it('removed a product from the props.cart when its delete button is clicked', () => {
  const setCartQuantity = jest.fn();
  const setCart = jest.fn();
  const setCartVisible = jest.fn();
  let cartInitLength = cart.length;
  act(() => {
    ReactDOM.createRoot(container).render(
      <BrowserRouter>
        <Nav
          cart={cart}
          cartQuantity={0}
          cartVisible={true}
          setCart={setCart}
          setCartQuantity={setCartQuantity}
          setCartVisible={setCartVisible}
        />
      </BrowserRouter>
    );
  });
  let delBtns = [...container.querySelectorAll('.cartitem__delete')];
  userEvent.click(delBtns[0]);
  expect(cart.length).toEqual(cartInitLength - 1);
  userEvent.click(delBtns[0]);
  expect(cart.length).toEqual(cartInitLength - 2);
});

it('increases the total price and quantity of a product when the button to increase its quantity is clicked', () => {
  const setCartQuantity = jest.fn();
  const setCart = jest.fn();
  const setCartVisible = jest.fn();

  act(() => {
    ReactDOM.createRoot(container).render(
      <BrowserRouter>
        <Nav
          cart={cart}
          cartQuantity={0}
          cartVisible={true}
          setCart={setCart}
          setCartQuantity={setCartQuantity}
          setCartVisible={setCartVisible}
        />
      </BrowserRouter>
    );
  });

  let productIncreaseQtyBtn = [
    ...container.querySelectorAll('.cartitem__qtymodifier.add'),
  ][1];
  let initialQtyValue = parseInt(
    [...container.querySelectorAll('.cartitem__qtyval')][1].textContent
      .split(': ')
      .pop()
  );
  let initialTotalValue = Number(
    [...container.querySelectorAll('.cartitem_itemtotal')][1].textContent
      .split(': ')
      .pop()
      .slice(0, -1)
  );
  let prodPrice = cart[1].product.salePrice;

  userEvent.click(productIncreaseQtyBtn);

  expect(initialQtyValue + 1).toEqual(cart[1].qty);
  expect((cart[1].qty * cart[1].product.salePrice).toFixed(2)).toEqual(
    (initialTotalValue + prodPrice).toString()
  );
});

it('decreases the total price and quantity of a product when the button to decrease its quantity is clicked', () => {
  const setCartQuantity = jest.fn();
  const setCart = jest.fn();
  const setCartVisible = jest.fn();

  act(() => {
    ReactDOM.createRoot(container).render(
      <BrowserRouter>
        <Nav
          cart={cart}
          cartQuantity={0}
          cartVisible={true}
          setCart={setCart}
          setCartQuantity={setCartQuantity}
          setCartVisible={setCartVisible}
        />
      </BrowserRouter>
    );
  });

  let productDecreaseQtyBtn = [
    ...container.querySelectorAll('.cartitem__qtymodifier.subtract'),
  ][1];
  let initialQtyValue = parseInt(
    [...container.querySelectorAll('.cartitem__qtyval')][1].textContent
      .split(': ')
      .pop()
  );
  let initialTotalValue = Number(
    [...container.querySelectorAll('.cartitem_itemtotal')][1].textContent
      .split(': ')
      .pop()
      .slice(0, -1)
  );
  let prodPrice = cart[1].product.salePrice;

  userEvent.click(productDecreaseQtyBtn);

  expect(initialQtyValue - 1).toEqual(cart[1].qty);
  expect((cart[1].qty * cart[1].product.salePrice).toFixed(2)).toEqual(
    (initialTotalValue - prodPrice).toFixed(2).toString()
  );

  userEvent.click(productDecreaseQtyBtn);
  userEvent.click(productDecreaseQtyBtn);

  expect(initialQtyValue - 3).toEqual(cart[1].qty);
  expect((cart[1].qty * cart[1].product.salePrice).toFixed(2)).toEqual(
    (initialTotalValue - prodPrice * 3).toFixed(2).toString()
  );
});

it('removes a product from props.products when its quantity is 1 and the decrease quantity button is clicked', () => {
  const setCartQuantity = jest.fn();
  const setCart = jest.fn();
  const setCartVisible = jest.fn();

  act(() => {
    ReactDOM.createRoot(container).render(
      <BrowserRouter>
        <Nav
          cart={cart}
          cartQuantity={0}
          cartVisible={true}
          setCart={setCart}
          setCartQuantity={setCartQuantity}
          setCartVisible={setCartVisible}
        />
      </BrowserRouter>
    );
  });

  let productDecreaseQtyBtn = [
    ...container.querySelectorAll('.cartitem__qtymodifier.subtract'),
  ][0];

  userEvent.click(productDecreaseQtyBtn);
  expect(cart[0].product.name).toEqual('Product 2');
  expect(cart[0].product.name).not.toEqual('Product 3');
  expect(cart[0].product.name).not.toEqual('Product 1');
});

it('decreases the total cart price when a product is removed props.products', () => {
  const setCartQuantity = jest.fn();
  const setCart = jest.fn();
  const setCartVisible = jest.fn();
  act(() => {
    ReactDOM.createRoot(container).render(
      <BrowserRouter>
        <Nav
          cart={cart}
          cartQuantity={0}
          cartVisible={true}
          setCart={setCart}
          setCartQuantity={setCartQuantity}
          setCartVisible={setCartVisible}
        />
      </BrowserRouter>
    );
  });
  let firstProdPrice = cart[0].qty * cart[0].product.salePrice;
  let productDecreaseQtyBtn = [
    ...container.querySelectorAll('.cartitem__qtymodifier.subtract'),
  ][0];
  let initialTotal = cart.reduce(
    (total, item) => total + item.qty * item.product.salePrice,
    0
  );

  userEvent.click(productDecreaseQtyBtn);

  let updatedTotal = cart
    .reduce((total, item) => total + item.qty * item.product.salePrice, 0)
    .toFixed(2);

  expect((initialTotal - firstProdPrice).toFixed(2).toString()).toEqual(
    updatedTotal
  );
});
/*
Tests:
    1. when a product is in props.cart it is visible in the cart - DONE
    2. when a product is deleted from the props.cart it is removed - DONE
    3. when a product's quantity is increased the total 
        // and quantity are increased - DONE
    4. when a product's quantity is decreased the total
        // and quantity are decreased - DONE
    5. when a product's quantity is 1 and it is decreased 
        // it is removed from the cart - DONE
    6. when there are 3 products in the cart and one is 
    removed the other 2 are still there and the total 
        // price is decreased
*/
