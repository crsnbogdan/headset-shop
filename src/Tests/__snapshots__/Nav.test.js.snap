// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches snapshot 1`] = `
<div
  className="nav__container"
>
  <nav
    className="nav items-center relative text-white flex px-28 py-6 justify-between"
  >
    <h3
      className="text-2xl text-yellow-300 font-semibold"
    >
      KB
      <span
        className="text-white font-light text-lg"
      >
        .shop
      </span>
    </h3>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge nav__sidebartoggle css-tzssek-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
    <div
      className="nav__links w-3/12 text-lg flex items-center justify-between"
      style={
        Object {
          "right": "-300px",
        }
      }
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge nav__close css-tzssek-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <a
        aria-current="page"
        className="nav__link cart active"
        href="/"
        onClick={[Function]}
      >
        home
      </a>
      <a
        className="nav__link"
        href="/shop"
        onClick={[Function]}
      >
        shop
      </a>
      <a
        className="nav__link relative"
        href="/cart"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-2psjec-MuiSvgIcon-root"
          data-testid="ShoppingCartIcon"
          focusable="false"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </a>
    </div>
    <div
      className="nav__cartcontainer p-2 bg-slate-100 absolute"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="nav__cartitem grid p-2 bg-white"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <img
          alt=""
          className="cartitem__img"
          src="https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg"
        />
        <div
          className="cartitem__info flex flex-col justify-between text-sm text-black"
        >
          <p
            className="text-black font-semibold cartitem_itemtotal"
          >
            <span
              className="font-normal"
            >
              Total: 
            </span>
            89.99
            $
          </p>
          <p
            className="text-sm"
          >
            Product 1(...)
          </p>
          <p
            className="cartitem__qtyval"
          >
            Quantity: 
            1
          </p>
        </div>
        <div
          className="cartitem__actions justify-between items-center flex flex-col"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__delete css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <div
            className="cartitem__qtymodifiers flex items-center "
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__qtymodifier add css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
            <p
              className="text-lg text-black font-semibold py-1 px-3 bg-slate-100"
            >
              1
            </p>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__qtymodifier subtract css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="nav__cartitem grid p-2 bg-white"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <img
          alt=""
          className="cartitem__img"
          src="https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/5053/5053501_sd.jpg"
        />
        <div
          className="cartitem__info flex flex-col justify-between text-sm text-black"
        >
          <p
            className="text-black font-semibold cartitem_itemtotal"
          >
            <span
              className="font-normal"
            >
              Total: 
            </span>
            359.96
            $
          </p>
          <p
            className="text-sm"
          >
            Product 2(...)
          </p>
          <p
            className="cartitem__qtyval"
          >
            Quantity: 
            4
          </p>
        </div>
        <div
          className="cartitem__actions justify-between items-center flex flex-col"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__delete css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <div
            className="cartitem__qtymodifiers flex items-center "
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__qtymodifier add css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
            <p
              className="text-lg text-black font-semibold py-1 px-3 bg-slate-100"
            >
              4
            </p>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__qtymodifier subtract css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="nav__cartitem grid p-2 bg-white"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <img
          alt=""
          className="cartitem__img"
          src="https://pisces.bbystatic.com/prescaled/500/500/image2/BestBuy_US/images/products/6298/6298657_sd.jpg"
        />
        <div
          className="cartitem__info flex flex-col justify-between text-sm text-black"
        >
          <p
            className="text-black font-semibold cartitem_itemtotal"
          >
            <span
              className="font-normal"
            >
              Total: 
            </span>
            159.98
            $
          </p>
          <p
            className="text-sm"
          >
            Product 3(...)
          </p>
          <p
            className="cartitem__qtyval"
          >
            Quantity: 
            2
          </p>
        </div>
        <div
          className="cartitem__actions justify-between items-center flex flex-col"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__delete css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <div
            className="cartitem__qtymodifiers flex items-center "
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__qtymodifier add css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
            <p
              className="text-lg text-black font-semibold py-1 px-3 bg-slate-100"
            >
              2
            </p>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cartitem__qtymodifier subtract css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </div>
        </div>
      </div>
      <p
        className="text-black nav__carttotal"
        style={Object {}}
      >
        Total:
         
        <span
          className="font-semibold"
        >
          609.93
          $
        </span>
      </p>
    </div>
  </nav>
</div>
`;
